From 7bd76288f7755bf467d6c9e7654665b34924f857 Mon Sep 17 00:00:00 2001
From: cixi <gengcx@spreadst.com>
Date: Thu, 26 Jul 2018 15:49:43 +0800
Subject: [PATCH] get_devnode.sh:add judgment if device not exist

on hikey board, the /dev/watchdog not exist, but the
keep continue test
Signed-off-by: cixi <gengcx@spreadst.com>
---
 testcases/ddt/scripts/common/get_devnode.sh | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/testcases/ddt/scripts/common/get_devnode.sh b/testcases/ddt/scripts/common/get_devnode.sh
index 8c36eda..14551c3 100755
--- a/testcases/ddt/scripts/common/get_devnode.sh
+++ b/testcases/ddt/scripts/common/get_devnode.sh
@@ -34,7 +34,12 @@ case $DEVICE_TYPE in
 		DEV_NODE="/dev/i2c-1"
 	;;
 	wdt)
-		DEV_NODE="/dev/watchdog"
+        if [ ! -e /dev/watchdog ];then
+            echo "Error: platform does not have wdt diveces"
+            exit 1
+        else
+		    DEV_NODE="/dev/watchdog"
+        fi
 	;;
         *)
 		DEV_NODE=`get_blk_device_node.sh "$DEVICE_TYPE"` || die "error getting $DEV_TYPE devnode"
-- 
1.9.1

